### ISSUE TITLE
The rule vue/require-direct-export does not support functional components

### ISSUE THREAD

**Issue by danyadev** [vuejs/eslint-plugin-vue#1197]
**Tell us about your environment**	
* **ESLint version:** 7.2.0
* **eslint-plugin-vue version:** 7.0.0-alpha.5
* **Node version:**	v12.16.3

**Please show your full configuration:**

```js
module.exports = {
  root: true,

  parserOptions: {
    sourceType: 'module'
  },

  env: {
    browser: true,
    node: true,
    es2020: true
  },

  plugins: ['import'],

  settings: {
    'import/resolver': 'webpack',
    'import/ignore': []
  },

  extends: ['plugin:vue/vue3-recommended'],

  rules: {
    // *eslint rules*
    // *import rules*
    // *other vue rules*

    'vue/require-direct-export': 'error'
  }
};
```

**What did you do?**	

```vue
<script>
import { h } from 'vue';

export default function Test() {
  return h('div', 'functional component content');
}
</script>
```

**What did you expect to happen?**
No errors

**What actually happened?**
```
4:1  error  Expected the component literal to be directly exported  vue/require-direct-export
```
